tensorflow/stream_executor/dnn.h

```diff
@@ -1318,6 +1312,14 @@ class DnnSupport {
       bool with_winograd_nonfused, int cc_major, int cc_minor,
       std::vector<AlgorithmDesc>* out_algorithms);
 
+  virtual bool GetMIOpenConvolveAlgorithms(
```

Comment:
@deven-amd I'm more than certainly this new interface would be challenged when we propose it upstream. Would it be a good idea to document its semantic here in the header to show the differences with `GetConvolveAlgorithms`?

Reply:
My hope is that this new interface will be accepted.

timshen has indicated in the past (cut-n-paste from feedback in PR 26018)

> But I still see GetConvolveAlgorithms() is changed to a union for both cuDNN and MIOpen; I imagine it should be MIOpenGetConvolveAlgorithms() as a pure addition of interface. I also didn't see any dispatch in cudnn_conv_runner.cc; I imagine it should call GetConvolveAlgorithms() or MIOpenGetConvolveAlgorithms(), depending on which target is built for.

that is where I stole then name from ;)

Reply:
w.r.t documenting the differences, yes I will add a description of the API and highlight the differences from the CUDA version.

Reply:
I'd advise to submit this interface change as a priority to seek for upstream feedback as it will impact follow-up feature (refer to discussion [here](https://github.com/tensorflow/tensorflow/pull/30884#issuecomment-515082917)). This will cause `ROCm` and `Cuda` implementation to diverge in `conv_algorithm_picker`.